### ISSUE TITLE
RFC: structure change

### ISSUE THREAD

**Issue by timdebruin** [larq/zoo#123]
In adding the R2B KD code to zoo I run into an issue that the needed code does not really fit the current directory structure. 

The training procedure requires:
- the models (`R2BNet()`,`StrongBaseLineNet()` ) which *do* fit into the current structure
- a `teacher-student` model class which contains the logic for tying together two models, adding teacher-student losses and metrics and does some model loading.
- the `losses` implementations (which I think would  be neater to have in a separate file)
- some (simplified) multi-stage experiment code and the experiment definitions (which might be a bit too bulky to put in `experiments.py`

In a chat with @koenhelwegen the following structure was proposed, please comment if disagree, as it would be a breaking change (without some `__init__` magic). 
```
larq_zoo/
  training/
    basic_experiments.py
    kd_experiments.py
    teacher_student.py
  core/
    datasets.py
    losses.py
    layers.py
  literature/
    r2bnet.py
      class R2BNet()
      class StrongBaseLineNet()
  sota/
```

**Comment by AdamHillier** [comment]
Sounds good to me, although I'm not convinced that this counts as a breaking change, as the functional interface to the models would not be any different.

**Comment by timdebruin** [comment]
Good point, noone would actually be importing things from `experiments.py`. So even better :+1:

**Comment by AdamHillier** [comment]
Closed by #127.
